### DISCUSSION TITLE
Enable N64 rumble on startup

### DISCUSSION THREAD

**Discussion by snowmanmcgee** [darthcloud/BlueRetro#848] [Q&A] [answered]
I have a RetroTime N64 Adapter and when I set the accessory to rumble on blueretro.io, it acts as though the rumble pak is plugged in after the console is turned on, instead of it already being plugged in as the console is turned on. This means that in Banjo-Kazooie the rumble does not work until the console is reset because a rumble pak only works in that game if it is already in the controller when the game is started. Is there any way to make the rumble functionality work so that the N64 sees the rumble pak as inserted when the console is turned on? Also, when I enable rumble as the accessory on blueretro.io, the adapter stops working entirely in Ocarina of Time. Is there a way to fix that too?

**Comment by darthcloud** [comment]
For that all you need to do is configure before hand BlueRetro output config accessories to be set to rumble and click save:
https://github.com/darthcloud/BlueRetro/wiki/BlueRetro-BLE-Web-Config-User-Manual#26---example-enable-rumble

When you power cycle the rumble will be there at boot.

I just tested now OoT real US cartridge and game boot no issue and rumble is functional.

Banjo is a picky game rumble detection-wise, it used to work but I can see it doesn't detect it anymore on cold power on.
As a work around wait a couple sec after power on and it reset on console, it will detect it.

I will investigate why it doesn't detect it at boot anymore.

**Reply by snowmanmcgee** [reply]
When I start up OoT with the accessory set to none, it turns on fine and a controller can sync and everything works perfectly normal. I can also enable rumble through button presses and the rumble works. However, if I set the accessory to rumble, the adapter's light starts holding green instead of blinking green. After holding green for a while, it then turns off for a while, and then it turns back on for a while, and continues to do that. When the adapter's light is acting like that, I can't connect a controller. Do you know about this happening? If it changes anything, my version of the game is the third NTSC version.

**Comment by darthcloud** [comment]
I found what was the issue for Banjo. The accessories change transition code what triggered at boot.

I got a fix in this beta:
https://github.com/darthcloud/BlueRetro/actions/runs/7122736568

The OoT issue look weird but maybe that build will hep with that aswell.

**Reply by snowmanmcgee** [reply]
Okay I just tried out the beta, and the issue with Banjo-Kazooie was completely fixed, but the issue with Ocarina of Time was not resolved. I also tried out OoT with the accessory set to the controller pak, and it worked perfectly fine. The only time the issue is there is when the accessory is set to the rumble pak. Another thing I noticed is that when resetting the console while the adapter is having the OoT issue, the N64 takes an unusually long time to reset and when it does, it says no controller is connected.

**Reply by darthcloud** [reply]
Ok this is very weird, can you give me as much detail as possible about you setup?

Does your n64 have any mod?
Do you know your n64 motherboard revision?
It's an original cartridge?

My OoT cart got B stamp on the back, so I think it the same as yours.

**Reply by snowmanmcgee** [reply]
My N64 is completely stock and I'm using official Nintendo S-Video cables with an official Nintendo power supply. I don't have the screwdriver for opening up my N64 so I don't know the motherboard revision, but the serial number starts with NS16. I know my cartridge is official and it also has the B on the back label. I also have the Expansion Pak in the N64.

**Comment by darthcloud** [comment, marked as answer]
I was was able to reproduce the OoT issue today somehow.

Try this new beta:
https://github.com/darthcloud/BlueRetro/actions/runs/7133883260

**Reply by snowmanmcgee** [reply]
I tested it and this beta works in Ocarina of Time and Banjo-Kazooie. Thanks for helping! :)
